synapse/api/auth.py

```diff
@@ -587,7 +587,10 @@ def _get_user_by_access_token(self, token):
     def _get_user_from_macaroon(self, macaroon_str):
         try:
             macaroon = pymacaroons.Macaroon.deserialize(macaroon_str)
-            self._validate_macaroon(macaroon)
+            self.validate_macaroon(
+                macaroon, "access",
+                [lambda c: c == "guest = true", lambda c: c.startswith("time < ")]
```

Comment:
The guest = true should always Just Work, is there a reason it needs to be pulled out?

And can you add a boolean verify_expiry to the validate_macaroon function, and pass that through to verify_expiry function as an arg, rather than passing in custom lambdas?